src/__tests__/BrahmosPureComponent.test.js

```diff
@@ -0,0 +1,70 @@
+/*
+Inspired by -
+https://github.com/facebook/react/blob/master/packages/react/src/__tests__/ReactPureComponent-test.js
+*/
+import Brahmos, { render } from "..";
+
+describe("BrahmosPureComponent", () => {
+  it("should render", () => {
+    let renders = 0;
+    class Component extends Brahmos.PureComponent {
+      constructor() {
+        super();
+        this.state = { type: "mushrooms" };
+      }
+      render() {
+        renders++;
+        return <div>{this.props.text[0]}</div>;
+      }
+    }
+
+    const container = document.createElement("div");
+    let text;
+    let component;
+
+    text = ["porcini"];
+    component = render(<Component text={text} />, container);
+    expect(container.textContent).toBe("porcini");
+    expect(renders).toBe(1);
+
+    text = ["morel"];
+    component = render(<Component text={text} />, container);
+    expect(container.textContent).toBe("morel");
+    expect(renders).toBe(2);
+
+    text[0] = "portobello";
+    component = render(<Component text={text} />, container);
+    expect(container.textContent).toBe("morel");
+    expect(renders).toBe(2);
+
+    // Setting state without changing it doesn't cause a rerender.
+    component.setState({ type: "mushrooms" }, () => {
```

Comment:
This should be moved on different spec

Reply:
@s-yadav Should I create a different test case for it?

Reply:
Yes. Prop change and state change are different actions which cause rerender, so both can be qualified as different behaviour. And it's better to write different specs for different behaviour.